packages/block-editor/src/components/use-setting/index.js

```diff
@@ -139,29 +142,39 @@ export default function useSetting( path ) {
 						false
 					)
 				) {
-					const candidateAtts =
+					const attributes =
 						select( blockEditorStore ).getBlockAttributes(
 							candidateClientId
 						);
-					const candidateResult =
-						get(
-							candidateAtts,
-							`settings.blocks.${ blockName }.${ normalizedPath }`
-						) ??
-						get( candidateAtts, `settings.${ normalizedPath }` );
-					if ( candidateResult !== undefined ) {
-						result = candidateResult;
+					result =
+						get( attributes, [
+							'settings',
+							'blocks',
+							blockName,
+							normalizedPath,
+						] ) ??
+						get( attributes, [ 'settings', normalizedPath ] );
+					if ( result !== undefined ) {
+						// Stop the search for more distant ancestors and move on.
```

Comment:
I'm surprised this does not have big noticeable effects on performance as per (probably because the test data does not have highly nested content?) https://github.com/WordPress/gutenberg/actions/runs/3083772483/jobs/4985121139

- firstBlock for post editor reduced by 100ms
- maxInserterSearch for post editor increased by 170ms
- firstBlock for site editor reduced by <100ms

There's also a big increase (300ms) for domContentLoad ond subsequent steps in the site editor, but I judge this unrelated to this PR.

Reply:
Now that the implementation is working, I'm more confused [by the results](https://github.com/WordPress/gutenberg/actions/runs/3097424144/jobs/5014082574). No impact in firstBlock and other content-related stats.